18: React App II - Other views

gabrielizalo opened the pull request:
# Summary

* 18: React App II - Other views
   - The Profile page is finished, and it loads the info from DB
   - In the App init it loads a dummy user
   - The logged user now can edit its data and their updates are saved in the DB


# Trello Card

https://trello.com/c/33Zbvi6T/7-react-app-ii-other-views


# Screenshot

<img width="1662" alt="other views 1" src="https://user-images.githubusercontent.com/656353/48920194-782b2080-ee65-11e8-8816-7fe2cfd9013c.png">
<img width="1662" alt="other views 2" src="https://user-images.githubusercontent.com/656353/48920197-7cefd480-ee65-11e8-8342-0ae4e21e66eb.png">
<img width="1662" alt="other views 3" src="https://user-images.githubusercontent.com/656353/48920199-7f522e80-ee65-11e8-9962-15419b7b1160.png">

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe"
- We now have a new Profile page design

gabrielizalo committed:
Merge branch '17-Routing-II-TopBar' into 18-React-App-II-Other-Views

gabrielizalo committed:
Merge branch '17-Routing-II-TopBar' into 18-React-App-II-Other-Views

gabrielizalo committed:
Merge remote-tracking branch 'origin/master' into 18-React-App-II-Other-Views

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Now the profile component gets the user's info from the API

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- The user id now is a constant

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fix of services and actions names

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- I have created a new service for users: Create. And a new user is
  created automatically every time the app is initialized.

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Nos the user can edit his data from profile page and that changes are
  updated in the DB thanks to the API

henryzarza commented on react/react-training/package.json:
```diff
@@ -18,5 +18,8 @@
     "eject": "react-scripts eject"
+  },
+  "devDependencies": {
+    "riek": "^1.1.0"
```
This isn't `devDependecies` move it to `dependencies` and i recommend you to do this without libraries

henryzarza commented on react/react-training/src/app/components/AppRoutes/index.js:
```diff
@@ -7,15 +7,23 @@ import Dashboard from '@screens/Dashboard';
 class AppRoutes extends Component {
+  componentDidMount() {
+    if (WE_ARE_ONLINE) {
+      const { userPost } = this.props;
```
I think that isn't necessary, you can use directly `this.props.userPost`

henryzarza commented on react/react-training/src/app/components/AppRoutes/index.js:
```diff
@@ -7,15 +7,23 @@ import Dashboard from '@screens/Dashboard';
+
   render() {
-    if (OFFLINE_WE_ARE) {
+    if (!WE_ARE_ONLINE) {
```
What happen if `WE_ARE_ONLINE ` is false? Change this for: `WE_ARE_ONLINE ? null : <Offline />`

henryzarza commented on react/react-training/src/app/components/AppRoutes/index.js:
```diff
@@ -34,11 +42,19 @@ class AppRoutes extends Component {
 AppRoutes.propTypes = {
-  userIsLogged: PropTypes.bool
+  userIsLogged: PropTypes.bool,
+  userPost: PropTypes.func
```
Is required?

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/TopBar/index.js:
```diff
@@ -9,8 +9,8 @@ import styles from './styles.scss';
   handleClick = () => {
-    const { logout } = this.props;
-    logout();
+    const { userLogout } = this.props;
```
You can use directly `this.props.userLogout()`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/index.js:
```diff
@@ -1,9 +1,51 @@
+class Profile extends Component {
+  componentDidMount() {
+    const userId = USER_ID;
+    const { userGet } = this.props;
```
The same here, i think that isn't necessary the destructuring

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/index.js:
```diff
@@ -1,9 +1,51 @@
+  }
+
+  render() {
+    const { userData } = this.props;
```
And here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:
```diff
@@ -0,0 +1,102 @@
+
+import { validateEmail } from '@utils';
+
+import photo from './assets/profile.jpg';
```
You never use the `assets/1.jpg` and `assets/2.jpg` so delete it

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:
```diff
@@ -0,0 +1,102 @@
+          <div className={styles.info}>
+            <p className={styles.text}>
+              <span className={styles.user}>Id:</span> {userData.id}
+              <br />
```
It isn't recommend to use this, please delete it

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:
```diff
@@ -0,0 +1,102 @@
+                shouldRemainWhileInvalid
+                classInvalid={styles.error}
+              />
+              <br />
```
The same here and the others

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/layout.js:
```diff
@@ -0,0 +1,102 @@
+}
+
+Layout.propTypes = {
+  userData: PropTypes.shape({
```
You're using this more than once, move it to `propTypes` constants file

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+
+.card {
+  align-content: space-around;
+  background-color: #757575;
```
Move this color definition to `colors.scss`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  align-content: space-around;
+  background-color: #757575;
+  border-radius: 10px;
+  box-shadow: 3px 3px 20px rgba(0, 0, 0, .5);
```
The same here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  border-radius: 10px;
+  box-shadow: 3px 3px 20px rgba(0, 0, 0, .5);
+  display: flex;
+  flex-direction: row;
```
This isn't necessary because is the default value of this property

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  margin: auto;
```
Change this for: `margin: 0 auto`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  justify-content: center;
+  margin: auto;
+  padding: 10px;
+  width: 500px;
```
Remember to use `max-width` property too

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  height: 150px;
+  width: 150px;
+
+  -webkit-animation: fadein 2s; /* Safari, Chrome and Opera > 12.1 */
```
This prefix isn't necessary because the sass compiler assigns it, so delete it all

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+}
+
+.text {
+  line-height: 1.3em;
```
Don't use `em` remember only use `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  to   { opacity: 1; }
+}
+
+@-moz-keyframes fadein {
```
The same of the prefixes here, delete it all

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  height: 4em;
+  width: 1em;
+
+  -webkit-animation: load1 1s infinite ease-in-out;
```
This isn't necessary

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+.loader,
+.loader:before,
+.loader:after {
+  background: #ffffff;
```
Use `$white` variable

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+.loader:before,
+.loader:after {
+  background: #ffffff;
+  height: 4em;
```
Remember to use only `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+.loader:after {
+  background: #ffffff;
+  height: 4em;
+  width: 1em;
```
The same here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+}
+
+.loader {
+  color: #ffffff;
```
Change this for `$white` variable

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  font-size: 11px;
+  margin: 88px auto;
+  position: relative;
+  text-indent: -9999em;
```
Remember only use `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  position: relative;
+  text-indent: -9999em;
+
+  -webkit-transform: translateZ(0);
```
The same of the prefixes

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  -ms-transform: translateZ(0);
+  transform: translateZ(0);
+
+  -webkit-animation-delay: -0.16s;
```
The same here and isn't necessary the break line

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  }
+
+  &:before {
+    left: -1.5em;
```
Change this for `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  &:before {
+    left: -1.5em;
+
+    -webkit-animation-delay: -0.32s;
```
The same of the prefixes

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  }
+
+  &:after {
+    left: 1.5em;
```
Change this for `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  }
+}
+
+@-webkit-keyframes load1 {
```
The same of the prefixes

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  80%,
+  100% {
+    box-shadow: 0 0;
+    height: 4em;
```
Don't use `em`, please change it for `px`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+    height: 4em;
+  }
+  40% {
+    box-shadow: 0 -2em;
```
I think that is necessary the color. What do you think?

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,151 @@
+  }
+  40% {
+    box-shadow: 0 -2em;
+    height: 5em;
```
Use `px`

henryzarza commented on react/react-training/src/app/screens/Login/index.js:
```diff
@@ -7,13 +7,13 @@ import LoginForm from './layout';
   componentDidMount() {
-    const { loginVerify } = this.props;
-    loginVerify();
+    const { userLoginVerify } = this.props;
```
Isn't necessary the destructuring

henryzarza commented on react/react-training/src/app/screens/Login/index.js:
```diff
@@ -7,13 +7,13 @@ import LoginForm from './layout';
   handleSubmit = values => {
-    const { login } = this.props;
-    login(values.username, values.password);
+    const { userLogin } = this.props;
```
The same here

henryzarza commented on react/react-training/src/constants/index.js:
```diff
@@ -6,4 +6,20 @@ export const ERROR_404 = 'Sorry. 404 page. Don`t cry.';
 export const OFFLINE_MESSAGE = 'Sorry. We are offline. Please come back in a few minutes.';
 
-export const OFFLINE_WE_ARE = false;
+export const WE_ARE_ONLINE = true;
```
I think that you can change this for: `navigator.onLine` so this variable makes sense. Check this [post](https://davidwalsh.name/detecting-online)

gabrielizalo replied:
@henryzarza ignoro este cambio según te expliqué para qué sirve esta constante.

henryzarza commented on react/react-training/src/redux/login/actions.js:
```diff
@@ -1,29 +1,135 @@
 export const actionsCreators = {
-  login: (username, password) => async dispatch => {
-    dispatch({ type: actionTypes.LOGIN });
+  userGet: userId => async dispatch => {
```
I think this action shouldn't go here because this isn't associated with `Login` component

henryzarza commented on react/react-training/src/redux/login/actions.js:
```diff
@@ -1,29 +1,135 @@
+    }
+  },
+
+  userPatch: (userId, newUserData) => async dispatch => {
```
The same here

henryzarza commented on react/react-training/src/redux/login/actions.js:
```diff
@@ -1,29 +1,135 @@
+    }
+  },
+
+  userPost: myUser => async dispatch => {
```
And here

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
 const Reducer = (state = initialState, action) => {
   switch (action.type) {
-    case actionTypes.LOGIN:
+    case actionTypes.USER_GET:
```
The same recommendation of the responsibility here

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
         ...action.payload
       };
-    case actionTypes.LOGIN_FAILURE:
+    case actionTypes.USER_GET_FAILURE:
```
And here

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
         ...action.payload
       };
-    case actionTypes.LOGIN_SUCCESS:
+    case actionTypes.USER_GET_SUCCESS:
```
And here

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_PATCH:
```
Here too

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_PATCH_FAILURE:
```
Here too

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_PATCH_SUCCESS:
```
Here too

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_POST:
```
Here too

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_POST_FAILURE:
```
Here too

henryzarza commented on react/react-training/src/redux/login/reducer.js:
```diff
@@ -3,32 +3,80 @@ import { actionTypes } from './actions';
+        ...state,
+        ...action.payload
+      };
+    case actionTypes.USER_POST_SUCCESS:
```
Here too

henryzarza commented on react/react-training/src/scss/variables/_colors.scss:
```diff
@@ -8,4 +8,5 @@ $green: #00FF00;
 $green-earls: #BED23C;
 $mine-shaft: #333132;
 $mine-shaft-dark: #222222;
```
Change this for `#222`

henryzarza commented on react/react-training/src/scss/variables/_colors.scss:
```diff
@@ -8,4 +8,5 @@ $green: #00FF00;
 $green-earls: #BED23C;
 $mine-shaft: #333132;
 $mine-shaft-dark: #222222;
+$red: #FF0000;
```
Change this for: `#F00`

henryzarza commented on react/react-training/src/services/UserService.js:
```diff
@@ -1,5 +1,8 @@
 
 export default {
-  login: (username, password) => api.post('/Users/login', { username, password })
+  userGet: userId => api.get(`/Users/${userId}`),
```
I recommend you to create `loginService` and `userService` to separate responsibilities

henryzarza reviewed (changes requested):
Good job, only a few comments 😄

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fix package.json dependencies

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fix minor details in App Routes

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fix minor details in TopBar
- Removed JPG files that are unused

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Fix layout texts

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Added missing max-width attributes.
- Fixing Profile component styles (ems, break lines...).

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Minor changes into Login and variables colors

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- I have created a new service for user needs and separated
  responsibilities from Login and Users
- Created new actions and reducers for User services and Store

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Moved the "User Data Shape" as a constant

gabrielizalo commented:
@henryzarza y @WFolini : Estamos re quete listossssss
Quedo atento y gracias

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Game/components/Board/components/Square/styles.scss:
```diff
@@ -10,6 +10,7 @@
   line-height: 34px;
   margin-right: -1px;
   margin-top: -1px;
+  max-width: 34px;
```
Change this for `100%`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/constants.js:
```diff
@@ -0,0 +1,13 @@
+import PropTypes from 'prop-types';
```
Move this to `constant` folder in the root of the project

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -9,13 +9,13 @@
   justify-content: center;
-  margin: auto;
+  margin: 0 auto;
+  max-width: 500px;
```
Change this for `100%`

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -29,23 +29,19 @@
+  animation: fadein 2s;
   border-radius: 50%;
   height: 150px;
+  max-width: 150px;
```
The same here

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -65,51 +61,25 @@
   animation: load1 1s infinite ease-in-out;
+  background-color: $white;
+  height: 48px;
+  max-width: 12px;
```
And here

henryzarza commented on react/react-training/src/app/screens/Offline/styles.scss:
```diff
@@ -13,5 +13,6 @@
 
 .image {
   height: 300px;
+  max-width: 700px;
```
And here

henryzarza reviewed (changes requested):
Very good job! Last changes and we are ready

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Changed all max-width attributes to 100%

gabrielizalo commented:
Readyyyy at 💯%
🦊

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Moved profile constant to app ones

wfolini commented on react/react-training/package.json:
```diff
@@ -11,6 +11,7 @@
     "redux-form": "^7.4.2",
     "react-router-dom": "^4.3.1",
     "redux-thunk": "^2.3.0",
+    "riek": "^1.1.0",
```
It isn't a good library. Let's keep it now, let's not make any changes, but let's keep it in mind for future projects

wfolini commented on react/react-training/src/constants/index.js:
```diff
@@ -6,4 +8,32 @@ export const ERROR_404 = 'Sorry. 404 page. Don`t cry.';
+  surname: 'Porras'
+};
+
+export const USER_DATA_SHAPE = {
```
Move this constant to `src/propTypes` folder

wfolini reviewed (changes requested):
Excellent! some comments and go

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Moved UserDataShape to PropTypes folder

gabrielizalo commented:
We are super hyper mega readyyyyyyy  🎉 🎉  🎉  🎉

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,106 @@
+  margin: 88px auto;
+  position: relative;
+  text-indent: -9999px;
+  transform: translateZ(0);
```
This is necessary?

henryzarza commented on react/react-training/src/app/screens/Dashboard/screens/Profile/styles.scss:
```diff
@@ -6,3 +6,106 @@
+  }
+}
+
+@keyframes load1 {
```
I recommend you don't animate this properties

henryzarza reviewed (dismissed):
Good job, check my comments before of the merge

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Removed unnecessary attribute

gabrielizalo committed:
Doing "ReactApp - Tic Tac Toe":
- Updating the loading animation using `opacity` attribute

gabrielizalo commented:
💥   This Pull Request is hot and ready   💥
